app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -70,4 +73,26 @@ class AccountController @Inject()(
     }
   }
 
+  def getExistingUndertaking: Action[AnyContent] = escAuthentication.async { implicit request =>
+    implicit val eori: EORI = request.eoriNumber
+    for {
+      undertakingOpt <- escService.retrieveUndertaking(eori)
+    } yield {
+      undertakingOpt match {
+        case Some(undertaking) =>
```

Comment:
Minor - you could have two case statements, the first with a guard condition instead. I tend to do this over nesting if..else but I can't really say it's much better.